jax/_src/cudnn/scaled_matmul_stablehlo.py

```diff
@@ -492,13 +492,17 @@ def quantize(x, config):
   elif config.mode == "nvfp4":
     assert config.scale_type == jnp.float8_e4m3fn
     assert config.global_scale.dtype == jnp.float32
+    SCALE_MAX = jnp.finfo(config.scale_type).max.astype(x.dtype)
 
-    scales = scales / config.global_scale
-    scales_q = jax.lax.optimization_barrier(scales.astype(jnp.float8_e4m3fn))
-    scaled_x = x / (scales_q.astype(jnp.float32) *
-                    config.global_scale).astype(x.dtype)
+    prev_amax = config.global_scale * (MAX * SCALE_MAX)
+    scales_q = jnp.clip(
+        (amax / prev_amax) * SCALE_MAX, 0, SCALE_MAX
```

Comment:
Are we sure this is correct? Shouldn't it be clip to `-SCALE_MAX` to `SCALE_MAX`?

Also, can we keep using the formula of `scales / config.global_scale` instead of `(amax / prev_amax) * SCALE_MAX`? This would be consistent with how we scale any target values?

Reply:
scales are non-negative value so [0, MAX] should be range.